### ISSUE TITLE
Convenience methods on output widget

### ISSUE THREAD

**Issue by jasongrout** [jupyter-widgets/ipywidgets#1724]
We should have convenience methods for appending stdout, stderr, and display_data messages to an output widget, so that this:

```python
out = Output()
out.outputs = out.outputs + ({'name': 'stdout', 'output_type': 'stream', 'text': 'hi\n'},)
```
becomes something like this:
```python
out.append_stdout('hi\n')
```
Likewise, also have an `out.append_stderr` and `out.append_display_data`.

**Comment by jasongrout** [comment]
(see https://github.com/jupyter-widgets/ipywidgets/issues/565#issuecomment-331522316)

**Comment by jasongrout** [comment]
Fixed by #1752.
